test/opexecuter.cpp

```diff
@@ -135,13 +135,41 @@ void OpExecuter::ExecuteOnTF() {
   DeactivateNGraph();
   ClientSession session(tf_scope_);
   ASSERT_EQ(Status::OK(), session.Run(sess_run_fetchoutputs_, &tf_outputs_));
+  for (int i = 0; i < tf_outputs_.size(); i++) {
+    NGRAPH_VLOG(5) << " TF op " << i << tf_outputs_[i].DebugString();
+  }
 }
 
 // Compares tf_outputs_ with ngraph_outputs_
 void OpExecuter::CompareNGraphAndTF() {
   ASSERT_EQ(tf_outputs_.size(), ngraph_outputs_.size());
   for (int i = 0; i < tf_outputs_.size(); i++) {
-    AssertTensorEquals(tf_outputs_[i], ngraph_outputs_[i]);
+    NGRAPH_VLOG(5) << " TF output " << i << tf_outputs_[i].DebugString() << endl;
+    NGRAPH_VLOG(5) << " NG output " << i << ngraph_outputs_[i].DebugString() << endl;
+    switch (expected_output_datatypes_[i]) {
+      case DT_FLOAT:
+        AssertTensorEquals<float>(
+            tf_outputs_[i], ngraph_outputs_[i],
+            [](float arg0, float arg1)  {  int x = abs(arg0 - arg1);
+                                          if(arg0 != 0 && arg1 != 0) {
+                                            return(abs(x/arg0) <= 0.001 && abs(x/arg1) <= 0.001);
+                                          } else if(arg0 == 0 && arg1 == 0) {
+                                            return true;
+                                          } else if(arg0 == 0 && arg1 != 0) {
+                                            return false; // because the abs(x/arg1) = 1 > 0.001
+                                          } else {
+                                            return false; // because the abs(x/arg0) = 1 > 0.001
+                                          }
+                                      
+                                        });
+        break;
```

Comment:
for the last 2 cases (one of the args is 0), we cannot always return false. 

Lets pick something from here:
https://en.wikipedia.org/wiki/Relative_change_and_difference

we want to: 
```
measure = |x-y|/f(x,y)
```

we can choose ```f = max``` or ```f(x,y) = |x+y|/2``` or ```f(x,y) = (|x|+|y|)/2```

I prefer the first one as [suggested](http://c-faq.com/fp/fpequal.html) by Doug Gwyn of ANSI C fame. In this case we have only 2 cases (both zero, everything_else):

```
bool both_are_zero = (x==0 && y==0);
return both_are_zero || ((|x-y|/max(|x|, |y|)) < tol) 
```